components/sync15/src/coll_state.rs

```diff
@@ -81,8 +83,12 @@ impl<'state> LocalCollStateMachine<'state> {
                             if ids.global == meta_global.sync_id
                                 && ids.coll == engine_meta.sync_id =>
                         {
+                            let coll_keys = CollectionKeys::from_encrypted_bso(
+                                self.global_state.keys.clone(),
+                                self.root_key,
+                            )?;
                             Ok(LocalCollState::Ready {
-                                key: self.global_state.keys.default.clone(),
+                                key: coll_keys.key_for_collection(name).clone(),
```

Comment:
Note that this was using `.default` before because I previously had the intention of dropping support for per-collection keys, but didn't follow through. It seemed odd to still have `key_for_collection()` exist but not be called.